src/Microsoft.AspNet.OData.Shared/Builder/ODataInstanceAnnotationContainer.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    /// <summary>
+    /// Default implementation for IODataInstanceAnnotationContainer
+    /// </summary>
+    public class ODataInstanceAnnotationContainer : IODataInstanceAnnotationContainer
+    {
+        private IDictionary<string, IDictionary<string, object>> instanceAnnotations;
+
+
+        /// <summary>
+        /// Default Constructor
+        /// </summary>
+        public ODataInstanceAnnotationContainer()
+        {
+            instanceAnnotations = new Dictionary<string, IDictionary<string, object>>();
+        }
+
+        /// <summary>
+        /// Method to Add an Instance Annotation to the CLR type
+        /// </summary>
+        /// <param name="annotationName">Name of Annotation</param>
+        /// <param name="value">Value of Annotation</param>
+        public void AddAnnotation(string annotationName, object value)
```

Comment:
>AddAnnotation [](start = 20, length = 13)

If we change GetAllAnnotations to GetAllTypeAnnotations, we should probably be consistent and change AddAnnotation to AddTypeAnnotation, and SetAnnotation to SetTypeAnnotation.